lib/ansible/modules/cloud/openstack/os_image.py

```diff
@@ -132,8 +164,12 @@ def main():
         ramdisk=dict(default=None),
         kernel=dict(default=None),
         properties=dict(type='dict', default={}),
+        volume=dict(default=None),
         state=dict(default='present', choices=['absent', 'present']),
     )
+
+    mutually_exclusive = [['filename', 'volume']],
```

Comment:
Is this supposed to be used somewhere?

``` python
In [1]: mutually_exclusive = [['filename', 'volume']],

In [2]: mutually_exclusive
Out[2]: ([['filename', 'volume']],)
```

This object is just a tuple with a nested list of lists. I suspect we need to load this into some function to make it actually mutually exclusive?

Reply:
Indeed, it needs to be passed to the AnsibleModule initialization to have any effect.
It's fixed now.